packages/trackerless-network/src/logic/neighbor-discovery/NeighborUpdateManager.ts

```diff
@@ -51,6 +52,7 @@ export class NeighborUpdateManager {
             if (res.removeMe) {
                 const nodeId = getNodeIdFromPeerDescriptor(neighbor.getPeerDescriptor())
                 this.config.neighbors.remove(nodeId)
+                this.config.connectionLocker.unlockConnection(neighbor.getPeerDescriptor(), this.config.streamPartId)
```

Comment:
Should we unlock connection in all cases where a node is removed the neighbor list? I.e. some of these:
- `NeighborUpdateManager:53`
- `NeighborUpdateRpcLocal:69`
- `ProxyClient:225`

Reply:
Unlocking was added to the first two in the list but the proxy needs a bigger refactor I'd rather include in another PR

Reply:
Maybe worth adding that proxy refactoring to "Future improvements" and/or create a Linear ticket?